modules/administration/pages/maintenance-windows.adoc

```diff
@@ -0,0 +1,118 @@
+
+Restricted actions significantly modify the client, and could potentially cause the client to stop running.
+Some examples of restricted actions are:
+
+* Package installation
+* Client upgrade
+* Service pack migration
+* Highstate application (for Salt clients)
+
+Unrestricted actions are minor actions that are considered safe and are unlikely to cause problems on the client.
+Some examples of unrestricted actions are:
+
+* Package profile update
+* Hardware refresh
+* Subscribing to software channels
+
+
+Before you begin, you must create an [path]``ical`` file for upload, or link to an [path]``ical`` file to create the calendar.
+You can create [path]``ical`` files in your preferred calendaring tool, such as Microsoft Outlook, Google Calendar, or KOrganizer.
+
+
+
+.Procedure: Uploading a New Maintenance Calendar
+. In the {productname} {webui}, navigate to menu:Schedule[Maintenance Windows > Calendars], and click btn:[Create].
+. In the [guimenu]``Calendar Name`` section, type a name for your calendar.
+. Either provide a URL to your [path]``ical`` file, or upload the file directly.
+. Click btn:[Create Calendar] to save your calendar.
+
+
+
+.Procedure: Creating a New Schedule
+. In the {productname} {webui}, navigate to menu:Schedule[Maintenance Windows > Schedules], and click btn:[Create].
+. In the [guimenu]``Schedule Name`` section, type a name for your schedule.
+. OPTIONAL: If your [path]``ical`` file contains events that apply to more than one schedule, check [guimenu]``Multi``.
+. Select the calendar to assign to this schedule.
+. Click btn:[Create Schedule] to save your schedule.
+
+
+
+.Procedure: Assigning a Schedule to a Client
+. In the {productname} {webui}, navigate to menu:Systems[Systems List], select the client to be assigned to a schedule, locate the [guimenu]``System Properties`` panel and click btn:[Edit These Properties].
+Alternatively, you can assign clients through the system set manager by navigating to menu:Systems[System Set Manager] and using the menu:Misc[Maintenance Windows] tab.
+. In the [guimenu]``Edit System Details`` page, locate the [guimenu]``Maintenance Schedule`` field, and select the name of the schedule to be assigned.
+. Click btn:[Update Properties] to assign the maintenance schedule.
+
+[NOTE]
+====
+When you assign a new maintenance schedule to a client, it is possible that the client might already have some restricted actions scheduled, and that these might now conflict with the new maintenance schedule.
+If this occurs, the {webui} will display an error and you will not be able to assign the schedule to the client.
+To resolve this, check the btn:[Cancel affected actions] option when you assign the schedule.
+This will cancel any previously scheduled actions that conflict with the new maintenance schedule.
+====
+
+
+When you have created your maintenance windows, you can schedule restricted actions, such as package upgrades, to be performed during the maintenance window.
```

Comment:
Just me feeling (feel free to ignore): I'm a bit tentative about the  "When you have created your maintenance windows"  phrase, could we use someting like "Now you can schedule restricted actions, such as package upgrades, to be performed during the maintenance window."?

Reply:
The only problem that I have with your suggestion is that we're assuming the reader knows what "now" refers to. We know that readers jump around documentation, so we can't assume they have both read the entire preceding content and performed all the preceding steps. By using the "when you ... you can" construction (which is becoming standard in our docs), we're giving some more signposts for the reader. I appreciate that it can sound a little clunky when you've read the whole thing though :wink: